credential/signing/jwt.go

```diff
@@ -19,55 +18,61 @@ const (
 	NonceProperty string = "nonce"
 )
 
-// SignVerifiableCredentialJWT is prepared according to https://www.w3.org/TR/vc-data-model/#jwt-encoding
+// SignVerifiableCredentialJWT is prepared according to https://w3c.github.io/vc-jwt/#version-1.1
 // which will soon be deprecated by https://w3c.github.io/vc-jwt/ see: https://github.com/TBD54566975/ssi-sdk/issues/191
 func SignVerifiableCredentialJWT(signer crypto.JWTSigner, cred credential.VerifiableCredential) ([]byte, error) {
 	if cred.IsEmpty() {
 		return nil, errors.New("credential cannot be empty")
 	}
+	if cred.Proof != nil {
+		return nil, errors.New("credential cannot already have a proof")
+	}
 
 	t := jwt.New()
-	expirationVal := cred.ExpirationDate
-	if expirationVal != "" {
-		var expirationDate = expirationVal
-		unixTime, err := rfc3339ToUnix(expirationVal)
-		if err != nil {
-			return nil, errors.Wrap(err, "could not convert expiration date to unix time")
-		}
-		expirationDate = string(unixTime)
-		if err := t.Set(jwt.ExpirationKey, expirationDate); err != nil {
+	if cred.ExpirationDate != "" {
+		if err := t.Set(jwt.ExpirationKey, cred.ExpirationDate); err != nil {
 			return nil, errors.Wrap(err, "could not set exp value")
 		}
+
+		// remove the expiration date from the credential
+		cred.ExpirationDate = ""
+	}
+
+	if err := t.Set(NonceProperty, uuid.New().String()); err != nil {
+		return nil, errors.Wrap(err, "setting nonce value")
 	}
 
 	if err := t.Set(jwt.IssuerKey, cred.Issuer); err != nil {
 		return nil, errors.Wrap(err, "could not set exp value")
 	}
+	// remove the issuer from the credential
+	cred.Issuer = ""
 
-	var issuanceDate = cred.IssuanceDate
-	if unixTime, err := rfc3339ToUnix(cred.IssuanceDate); err == nil {
-		issuanceDate = string(unixTime)
-	} else {
-		// could not convert iat to unix time; setting to present moment
-		issuanceDate = strconv.FormatInt(time.Now().UTC().UnixNano(), 10)
+	if err := t.Set(jwt.IssuedAtKey, cred.IssuanceDate); err != nil {
+		return nil, errors.Wrap(err, "could not set iat value")
 	}
-
-	if err := t.Set(jwt.NotBeforeKey, issuanceDate); err != nil {
+	if err := t.Set(jwt.NotBeforeKey, cred.IssuanceDate); err != nil {
 		return nil, errors.Wrap(err, "could not set nbf value")
 	}
+	// remove the issuance date from the credential
+	cred.IssuanceDate = ""
```

Comment:
This is to spec?

Reply:
yes, strange to me too - see the spec text below and examples https://w3c.github.io/vc-jwt/#jwt-decoding

> In the example above, vc does not contain the id [property](https://www.w3.org/TR/vc-data-model/#dfn-property) because the JWT encoding uses the jti attribute to represent a unique identifier. The sub attribute encodes the information represented by the id [property](https://www.w3.org/TR/vc-data-model/#dfn-property) of credentialSubject